src/awkward/_connect/rdataframe/from_rdataframe.py

```diff
@@ -171,6 +171,10 @@ def cpp_fill_function(depth):
                 raise ak._errors.wrap_error(
                     TypeError(f'"{col}" column\'s type "{col_type}" is not supported.')
                 )
+            elif form_str == "awkward type":
+                raise ak._errors.wrap_error(
+                    RuntimeError("this code should not be reached.")
```

Comment:
One thing: we use `AssertionError` to denote unreachable code.

```suggestion
                    AssertionError("this code should not be reached.")
```